langchain/llms/gooseai.py

```diff
@@ -0,0 +1,143 @@
+"""Wrapper around GooseAI API."""
+import logging
+from typing import Any, Dict, List, Mapping, Optional
+
+from pydantic import BaseModel, Extra, Field, root_validator
+
+from langchain.llms.base import LLM
+from langchain.utils import get_from_dict_or_env
+
+logger = logging.getLogger(__name__)
+
+
+class GooseAI(LLM, BaseModel):
+    """Wrapper around OpenAI large language models.
+
+    To use, you should have the ``openai`` python package installed, and the
+    environment variable ``GOOSEAI_API_KEY`` set with your API key.
+
+    Any parameters that are valid to be passed to the openai.create call can be passed
+    in, even if not explicitly saved on this class.
+
+    Example:
+        .. code-block:: python
+            from langchain import GooseAI
+            gooseai = GooseAI(model_name="gpt-neo-20b")
+
+    """
+
+    client: Any
+
+    model_name: str = "gpt-neo-20b"
+    """Model name to use"""
+
+    temperature: float = 0.7
+    """What sampling temperature to use"""
+
+    max_tokens: int = 256
+    """The maximum number of tokens to generate in the completion.
+    -1 returns as many tokens as possible given the prompt and
+    the models maximal context size."""
+
+    top_p: float = 1
+    """Total probability mass of tokens to consider at each step."""
+
+    min_tokens: int = 1
+    """The minimum number of tokens to generate in the completion."""
+
+    frequency_penalty: float = 0
+    """Penalizes repeated tokens according to frequency."""
+
+    presence_penalty: float = 0
+    """Penalizes repeated tokens."""
+
+    n: int = 1
+    """How many completions to generate for each prompt."""
+
+    model_kwargs: Dict[str, Any] = Field(default_factory=dict)
+    """Holds any model parameters valid for `create` call not explicitly specified."""
+
+    logit_bias: Optional[Dict[str, float]] = Field(default_factory=dict)
+    """Adjust the probability of specific tokens being generated."""
```

Comment:
I think it'll be great to add the goose specific parameters (https://goose.ai/docs/api/completions) they've implemented, like `typical_p`. This, in addition to their low cost, are a big advantage to using their service 😃 

![image](https://user-images.githubusercontent.com/29989939/218441872-90e473fa-79e9-4ba3-bf5a-c85eb7cfcf97.png)

Or maybe just make a note that's easily visible to check goose docs and let them be kwargs. What do you think?

Reply:
I think `model_kwargs` would cover this?

Reply:
Yes sir 😃